Services.Test/Storage/DocumentDb/DocumentDbRecordTest.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Threading;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Storage.DocumentDb;
+using Xunit;
+
+namespace Services.Test.Storage.DocumentDb
+{
+    public class DocumentDbRecordTest
+    {
+        private DocumentDbRecord target;
+        private static long Now => DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+
+        public DocumentDbRecordTest()
+        {
+            this.target = new DocumentDbRecord();
+        }
+
+        [Fact]
+        public void ItCanUnlockForCurrentOwner()
+        {
+            // Arrange
+            var ownerId = "foo";
+            var ownerType = "bar";
+            var durationMs = 100;
+
+            // Act
+            this.target.Lock(ownerId, ownerType, durationMs);
+
+            // Assert
+            Assert.True(this.target.CanUnlock(ownerId, ownerType));
+        }
+
+        [Fact]
+        public void ItCanUnlockAfterExpiration()
+        {
+            // Arrange
+            var ownerId = "foo";
+            var ownerType = "bar";
+            var durationSeconds = 1;
+
+            // Act
+            this.target.Lock(ownerId, ownerType, durationSeconds);
+            Thread.Sleep(durationSeconds * 2 * 1000);
```

Comment:
we usually avoid locking tests for more than few msecs, because in time this delay accumulates and makes the whole build process longer than needed. Unit tests are supposed to be super-super fast, with very few exceptions. That said, this is a good test and working around the need of "sleeping" would need some refactoring, to isolate the code from the real CPU clock, and injecting a fake clock. i.e. we'd need the ability to manipulate time and define what "Now" is. So my suggestion is to leave a TODO note for the time being, so that we can come back to this later.

Reply:
Added a TODO.